### ISSUE TITLE
Sink ignores inputSpecs in configuration yaml on create

### ISSUE THREAD

**Issue by evanhlavaty** [apache/pulsar#4787]
**Describe the bug**
When creating a sink with an accompanying configuration yaml, I want to increase the receiverQueueSize so I have the following set in my configuration yaml

```
inputSpecs:
  sample_topic_name:
    receiverQueueSize: 1000000
```

however, my inputSpecs configuration settings are ignored after creation of the sink and receiverQueueSize is set to the default of 1000

**To Reproduce**
- **Create sink with pulsar admin cli**
[solr_kerberos_io-2.3.2.yml](https://github.com/apache/pulsar/files/3423473/sink_config.txt)
```
pulsar-admin --admin-url https://sample-admin:6751 sink create \
--sink-config-file /data/connectors/solr_kerberos_io-2.3.2.yml
```

`"Created Successfully"`

- **Confirm sink settings**

```
pulsar-admin --admin-url https://sample-admin:6751 sink get \
--tenant public \
--namespace default \
--name solr_kerberos

{
  "tenant": "public",
  "namespace": "default",
  "name": "solr_kerberos",
  "className": "org.apache.pulsar.io.solrcloud.SolrCloudSink",
  "inputSpecs": {
    "sample_topic_name": {
      "isRegexPattern": false
    }
  },
  "configs": {
    "solrUrl": "http://sample-solr:8983/solr",
    "loginConfigurationLocation": "/data/connectors/solr_kerberos_io_jaas.conf",
    "collectionName": "sample_collection",
    "documentQueueSize": "5000.0",
    "concurrentThreadCount": "25.0"
  },
  "parallelism": 3,
  "processingGuarantees": "ATLEAST_ONCE",
  "retainOrdering": false,
  "autoAck": true,
  "archive": "builtin://solr_kerberos"
}
```

**Expected behavior**
I expect the receiverQueueSize to be set to what I define it to be in the configuration yaml and not be overridden and/or ignored.

**Desktop (please complete the following information):**
NAME="Oracle Linux Server"
VERSION="7.6"
ID="ol"
VARIANT="Server"
VARIANT_ID="server"
VERSION_ID="7.6"
PRETTY_NAME="Oracle Linux Server 7.6"

**Additional context**
Pulsar Version 2.3.2. Sink is running on a cluster of function workers/brokers/proxies. When creating a function using the same inputSpecs yaml entry the receiverQueueSize is set correctly. This issue only seems to be related to sinks

**Comment by wolfstudy** [comment]
Hello, according to the information you provided, we have reproduced your process, and the `receiverQueueSize` can be assigned normally.

solr-config.yml as follows:

```
tenant: public
namespace: default
name: solr_kerberos
inputSpecs:
  sample_topic_name:
    receiverQueueSize: 1000000
configs:
  solrUrl: http://sample-solr:8983/solr
  loginConfigurationLocation: "/data/connectors/solr_kerberos_io_jaas.conf"
  collectionName: sample_collection
  documentQueueSize: 5000
  concurrentThreadCount: 25
parallelism: 3
processingGuarantees: ATLEAST_ONCE
retainOrdering: false
autoAck: true
```

### To Reproduce

```
➜  pulsar git:(master) ✗docker run -it -d  -p 8983:8983 --name pulsar-solr-docker solr solr-precreate gettingstarted

➜  pulsar git:(master) ✗ ./bin/pulsar-admin sink create --sink-config-file connectors/solr-config.yaml -a pulsar-io/solr/target/pulsar-io-solr-2.4.0.nar
"Created successfully"
➜  pulsar git:(master) ✗ ./bin/pulsar-admin sink get --tenant public --namespace default --name solr_kerberos
{
 "tenant": "public",
 "namespace": "default",
 "name": "solr_kerberos",
 "className": "org.apache.pulsar.io.solr.SolrGenericRecordSink",
 "inputSpecs": {
   "sample_topic_name": {
     "isRegexPattern": false,
     "receiverQueueSize": 1000000
   }
 },
 "configs": {
   "solrUrl": "http://localhost:8983/solr",
   "documentQueueSize": 5000,
   "concurrentThreadCount": 25,
   "collectionName": "sample_collection"
 },
 "parallelism": 3,
 "processingGuarantees": "ATLEAST_ONCE",
 "retainOrdering": false,
 "autoAck": true
}
```

**Comment by evanhlavaty** [comment]
Is there anyway you can confirm there is an issue in 2.3.2 or not, I will not be able to upgrade to the latest version right away and I don't want to keep trouble shooting in 2.3.2 if its not doable.

**Comment by evanhlavaty** [comment]
> Is there anyway you can confirm there is an issue in 2.3.2 or not, I will not be able to upgrade to the latest version right away and I don't want to keep trouble shooting in 2.3.2 if its not doable.

I was able to confirm this was a 2.3.2 issue in SinkConfigUtils -> convertFromDetails method completely ignores the receiverQueueSize

**Comment by wolfstudy** [comment]
Yes, the issue is fixed in [https://github.com/apache/pulsar/pull/4091](https://github.com/apache/pulsar/pull/4091). 

Unfortunately, he was not released to the 2.3.2 branch. This issue was fixed in 2.4.0. If you need it, you can cherry-pick this fix to branch-2.3 and put it in the 2.3.3 tag.
